File: fixtures/dotnet/j16/tests/Fixture.cs
// Supporting types for j16 tests
// The model writes DataPipeline.cs with the AsyncDataPipeline class implementation
// Note: The async interfaces (IAsyncDatabase, IAsyncEnricher, IAsyncCache, IAsyncNotifier)
// are defined in the test file itself (DataPipelineTests.cs)

public class UserData
{
    public string Id { get; set; } = "";
    public string Email { get; set; } = "";
    public string Name { get; set; } = "";
    public bool Enriched { get; set; }
}

public class PipelineResult
{
    public string UserId { get; set; } = "";
    public string Status { get; set; } = "";
    public bool Cached { get; set; }
    public bool Notified { get; set; }
}

File: fixtures/java/j16/tests/src/main/java/Fixture.java
// Supporting classes for j16 DataPipeline tests
// These interfaces and classes are used by both the model-generated DataPipeline and the test suite

import java.util.function.BiConsumer;

interface Database {
    void findUser(String userId, BiConsumer<Exception, UserData> callback);
}

interface Enricher {
    void enrich(UserData user, BiConsumer<Exception, UserData> callback);
}

interface Cache {
    void set(String key, UserData value, BiConsumer<Exception, Void> callback);
}

interface Notifier {
    void notify(String email, String message, BiConsumer<Exception, Void> callback);
}

class UserData {
    String id;
    String email;
    String name;
    boolean enriched;

    public UserData(String id, String email, String name) {
        this.id = id;
        this.email = email;
        this.name = name;
        this.enriched = false;
    }

    public String getId() { return id; }
    public String getEmail() { return email; }
    public String getName() { return name; }
    public boolean isEnriched() { return enriched; }
    public void setEnriched(boolean enriched) { this.enriched = enriched; }
}

class PipelineResult {
    String userId;
    String status;
    boolean cached;
    boolean notified;

    public PipelineResult(String userId, String status, boolean cached, boolean notified) {
        this.userId = userId;
        this.status = status;
        this.cached = cached;
        this.notified = notified;
    }

    public String getUserId() { return userId; }
    public String getStatus() { return status; }
    public boolean isCached() { return cached; }
    public boolean isNotified() { return notified; }
}
